processor/routingprocessor/router.go

```diff
@@ -56,6 +52,11 @@ func newRouter[E component.Exporter](config Config, logger *zap.Logger) router[E
 	}
 }
 
+type routedSignal[E component.Exporter, S plog.Logs | pmetric.Metrics | ptrace.Traces] struct {
```

Comment:
Is there an advantage of this over the previous version (wrapping this in a type)? I find the dedicated type easier to read.

Reply:
I reverted back to the declaration of the `signal` type constraint.